### ISSUE TITLE
Removing existing inline styles.

### ISSUE THREAD

**Issue by CrashyBang** [jlmakes/scrollreveal#372]
I have the following code applied to an image on window load:

```js
element.setAttribute('style', `top: calc(100% - ${ calculatedNumberOfPixels })`)
```

Now once ScrollReveal kicks in, it removes this style. As per #251, I believe ScrollReveal should preserve existing styles?

Thanks!

- Ubuntu Minimal (XFCE) 16.10
- Chrome: 58.0.3029.110 (64-bit)
- `scrollreveal@3.3.6`

**Comment by jlmakes** [comment]
From the [Mozilla Developer Network](https://developer.mozilla.org/en-US/docs/Web/Reference/Events/DOMContentLoaded):

> The DOMContentLoaded event is fired when the document has been completely loaded and parsed, without waiting for stylesheets, images, and subframes to finish loading (**the load event can be used to detect a fully-loaded page**).

So, it really sounds like ScrollReveal is running after `DOMContentLoaded` but before window’s `load` event. So when ScrollReveal does it’s setup (and capturing of inline styles), your code to apply `top: calc(...)` hasn’t even ran yet.

When it _does_ run (after images, stylesheets and fonts finish loading), you’re actually overwriting the styles ScrollReveal generated and attached to the element.

You won’t see that though, because ScrollReveal will re-apply its styles when the event handler is triggered (on scroll or resize)—overwriting your `top: calc(...)`.

**So basically, your `reveal()` calls should be made after (or inside) your window `load` handler.**

**Comment by CrashyBang** [comment]
Hey @jlmakes,

Ahh I see that makes sense!

Will fix that now!

Cheers.
